main/CMakeLists.txt

```diff
@@ -20,7 +20,7 @@ foreach(file ${www_files})
     get_filename_component(www_file_dir ${build_dir}/www/${file}.gz DIRECTORY)
     file(MAKE_DIRECTORY ${www_file_dir})
     add_custom_command(OUTPUT ${build_dir}/www/${file}.gz
-        COMMAND gzip -c ${PROJECT_DIR}/www/${file} > ${build_dir}/www/${file}.gz
+        COMMAND tar -czf ${build_dir}/www/${file}.gz ${PROJECT_DIR}/www/${file}
```

Comment:
Hey, does this work with the web interface? A gzipped tar isn't the same as a gzipped file as it includes some tar headers which shouldn't be sent as a response by the web server.

Reply:
Ah, you are right, it does not work. I did not try the web frontend before, was just happy I got it compiling on windows... 🤔

Reply:
I was looking for alternatives at the time. Particularly, a Python gzip module that seems to already be installed with ESP-IDF, but that didn't have the option of specifying the output file, or something like that

Reply:
Actually, looking at the code of that module now, you might be able to get away with:
```cmake
COMMAND ${python} -m gzip < ${PROJECT_DIR}/www/${file} > ${build_dir}/www/${file}.gz`
```
If it does work, I'd appreciate a separate pull request for it or, at least, a separate commit.
Thanks!

Reply:
I'll give it a try and return with an extra PR. 
Have some other changes anyways.

Thanks for the cool Project !!

Reply:
Still produces the same error: :-/

httpd_txrx: httpd_resp_send_err: 431 Request Header Fields Too Large - Header fields are too long for server to interpret